tests/test_ote_training.py

```diff
@@ -0,0 +1,613 @@
+        labels_list = self.labels_schema.get_labels(False)
+        self.dataset.set_project_labels(labels_list)
+        print(f'train dataset: {len(self.dataset.get_subset(Subset.TRAINING))} items')
+        print(f'validation dataset: {len(self.dataset.get_subset(Subset.VALIDATION))} items')
+
+        logger.debug('Load model template')
+        self.model_template = parse_model_template(self.template_file_path)
+
+        hyper_parameters = self.model_template.hyper_parameters.data
+        set_values_as_default(hyper_parameters)
+
+        logger.debug('Setup environment')
+        params = create(hyper_parameters)
+        logger.debug('Set hyperparameters')
+        params.learning_parameters.num_iters = self.num_training_iters
+        if self.num_training_iters < 20:
+            num_checkpoints = 2
+        elif self.num_training_iters < 1000:
+            num_checkpoints = 10
+        else:
+            num_checkpoints = 30
+
+        #### TODO: fixed parameter, delete this
+        params.learning_parameters.batch_size = 2
+
+        params.learning_parameters.num_checkpoints = num_checkpoints
+
+        self.environment, self.task = self._create_environment_and_task(params,
+                                                                        self.labels_schema,
+                                                                        self.model_template)
+
+        logger.debug('Train model')
+        self.output_model = ModelEntity(
+            self.dataset,
+            self.environment.get_model_configuration(),
+            model_status=ModelStatus.NOT_READY)
+
+        self.copy_hyperparams = deepcopy(self.task._hyperparams)
+
+        self.task.train(self.dataset, self.output_model)
+        logger.info(f'performance={self.output_model.performance}')
+
+    def get_training_performance(self):
+        return getattr(self.output_model, 'performance', None)
+
+    def run_ote_training_once(self, data_collector):
+        if self.was_training_run and self.stored_exception_training:
+            logger.warn('In function run_ote_training_once: found that previous call of the function '
+                        'caused exception -- re-raising it')
+            raise self.stored_exception_training
+
+        if not self.was_training_run:
+            try:
+                self._run_ote_training()
+                self.was_training_run = True
+            except Exception as e:
+                self.stored_exception_training = e
+                self.was_training_run = True
+                raise e
+
+        training_performance = self.get_training_performance()
+
+        score_name, score_value = performance_to_score_name_value(training_performance)
+        if score_name:
+            data_collector.log_final_metric('training_performance/' + score_name, score_value)
+        else:
+            logger.warning(f'WARNING: Cannot get training performance')
```

Comment:
Why is this not an error?

BTW, IMO, no need to print "WARNINIG" inside the `warning`.

Reply:
Do you think that the test should fail if we cannot take model performance?

Reply:
Removed 'WARNING' from the message

Reply:
🤷‍♂️

Reply:
Ok, I will make the code a bit more strict.

Now if we cannot get the training performance, or the training performance score is invalid, the function `_run_ote_training` will raise an exception (and the function `run_ote_training_once` will re-raise the exception for the second call -- in the same way as for other exceptions)

Reply:
Done